extensions/json-jackson/src/main/java/io/deephaven/json/jackson/IntMixin.java

```diff
@@ -0,0 +1,154 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.json.jackson;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import io.deephaven.base.ArrayUtil;
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.json.IntOptions;
+import io.deephaven.json.jackson.IntValueProcessor.ToInt;
+import io.deephaven.qst.type.Type;
+import io.deephaven.util.QueryConstants;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import static io.deephaven.util.type.ArrayTypeUtils.EMPTY_INT_ARRAY;
+
+final class IntMixin extends Mixin<IntOptions> implements ToInt {
+
+    public IntMixin(IntOptions options, JsonFactory factory) {
+        super(factory, options);
+    }
+
+    @Override
+    public int numColumns() {
+        return 1;
+    }
+
+    @Override
+    public Stream<List<String>> paths() {
+        return Stream.of(List.of());
+    }
+
+    @Override
+    public Stream<Type<?>> outputTypesImpl() {
+        return Stream.of(Type.intType());
+    }
+
+    @Override
+    public ValueProcessor processor(String context, List<WritableChunk<?>> out) {
+        return new IntValueProcessor(out.get(0).asWritableIntChunk(), this);
+    }
+
+    @Override
+    public int parseValue(JsonParser parser) throws IOException {
```

Comment:
At the very least, some amount of switching is "necessary" to differentiate between the generic JSON types.

If we really really wanted to, we could have separate implementations for all combinations of configuration values to remove the need to check whether the type is allowed. In the case of int, we allow the user to finely configure INT, DECIMAL, STRING, NULL, and MISSING, so that would be 2^5 == 32 different implementations. For maintainability and sanity, I think the current approach of a single impl is much better. I'm also claiming that the added cost of checking "does not matter".